src/compiler/transformers/module/es6.ts

```diff
@@ -34,19 +34,53 @@ namespace ts {
                     return visitImportSpecifier(<ImportSpecifier>node);
                 case SyntaxKind.ExportAssignment:
                     return visitExportAssignment(<ExportAssignment>node);
+                case SyntaxKind.ExportDeclaration:
+                    return visitExportDeclaration(<ExportDeclaration>node);
+                case SyntaxKind.NamedExports:
+                    return visitNamedExports(<NamedExports>node);
+                case SyntaxKind.ExportSpecifier:
+                    return visitExportSpecifier(<ExportSpecifier>node);
             }
 
             return node;
         }
 
-        function visitExportAssignment(node: ExportAssignment): ExportDeclaration {
+        function visitExportAssignment(node: ExportAssignment): ExportAssignment {
             if (node.isExportEquals) {
                 return undefined; // do not emit export equals for ES6
             }
             const original = getOriginalNode(node);
             return nodeIsSynthesized(original) || resolver.isValueAliasDeclaration(original) ? node: undefined;
         }
 
+        function visitExportDeclaration(node: ExportDeclaration): ExportDeclaration {
+            if (!node.exportClause) {
+                return node; // export * is always emitted

```

Comment:
@mhegazy  I think we should though in the old emitter we are doing this only for downlevel compilation that does not have native support for `export *`